src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -208,30 +208,35 @@ public void StartConnectAsync()
             if (this.ConnectedPeers.Count >= this.MaximumNodeConnections)
                 return;
 
+            PeerAddress peerAddress = this.FindPeerToConnectTo();
+            if (peerAddress == null)
+            {
+                Task.Delay(TimeSpans.TenSeconds.Milliseconds).Wait(this.nodeLifetime.ApplicationStopping);
+                return;
+            }
+
             NetworkPeer peer = null;
 
             try
             {
-                PeerAddress peerAddress = this.FindPeerToConnectTo();
-                if (peerAddress == null)
-                {
-                    Task.Delay(TimeSpans.TenSeconds.Milliseconds).Wait(this.nodeLifetime.ApplicationStopping);
-                    return;
-                }
-
                 using (var timeoutTokenSource = CancellationTokenSource.CreateLinkedTokenSource(this.nodeLifetime.ApplicationStopping))
                 {
-                    timeoutTokenSource.CancelAfter(5000);
-
                     this.peerAddressManager.PeerAttempted(peerAddress.NetworkAddress.Endpoint, this.dateTimeProvider.GetUtcNow());
 
                     var clonedConnectParamaters = this.CurrentParameters.Clone();
+                    timeoutTokenSource.CancelAfter(5000);
                     clonedConnectParamaters.ConnectCancellation = timeoutTokenSource.Token;
 
                     peer = await this.networkPeerFactory.CreateConnectedNetworkPeerAsync(this.network, peerAddress.NetworkAddress, clonedConnectParamaters).ConfigureAwait(false);
                     await peer.VersionHandshakeAsync(this.Requirements, timeoutTokenSource.Token).ConfigureAwait(false);
                 }
             }
+            catch (OperationCanceledException timeout)
+            {
+                this.logger.LogInformation("Peer {0} connection timeout.", peerAddress.NetworkAddress.Endpoint);
```

Comment:
i don't think this should be info level

Reply:
OK this was where mentioned earlier in the slack channel, that it might be beneficial to the user to see that we are in fact trying to connect? We had this issue where the node was running for minutes without any connections, making it appear that it wasnt doing anything, where in actual fact it WAS trying to connect but just not being successful.

I don't really mind either way? WDYT?

Reply:
if you log it @ DEBUG level then using `-debug=1` without any additional settings, you would get this information being logged into `node.txt`, this seems to me simple enough for such scenarios to be checked

Reply:
Ok cool!